### PR TITLE
[Bootcamp] Week 7 Exercise (Stephen)

### PR DESCRIPTION
**Pull request by muthash**
## [GenAI Bootcamp] Fine-Tuning an open-source model
I tested again using updated Learning rate and warmup ratios configuration, but the results actually became worse:
EPOCHS = 1
BATCH_SIZE = 16  # on an A100 box
LEARNING_RATE = 3e-4
WARMUP_RATIO = 0.06

### Solution Snippets:

Link - https://colab.research.google.com/drive/1CAF3VcNvZs3Eex71TMX3A5exw2Bw4HUF?usp=sharing

Screenshot 2025-10-29 at 11 08 01

### CODE REVIEW

**Comment by ed-donner** [comment]
Haha fine-tuning can be frustrating! But nicely done anyway.. thanks Stephen

**Comment by ed-donner** [comment]
Oh wait the training and testing files are massive.. would you be able to replace this with a link to colab? It's 15,000 lines to check in..
I'll mark you comlpete in the sheet, but please update this PR when you get a chance.. thank you

**Comment by muthamastphn** [comment]
@ed-donner, the PR is updated with Colab links

**Comment by ed-donner** [comment]
fab - thanks so much - very nice
